runtime/parachains/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Runtime modules for parachains code.
+//!
+//! It is crucial to include all the modules from this crate in the runtime, in
+//! particular the `Initializer` module, as it is responsible for initializing the state
+//! of the other modules.
+
+mod configuration;
```

Comment:
@coriolinus These modules are all private for now, because I haven't exposed the public API yet. But the initializer lets us set some invariants for those other modules.

Reply:
I've talked with @shawntabrizi about something like "batch" modules, since his question was (paraphrased) "if these modules are tightly coupled, why not just have one module?".

It's a good question, and the answer I had was just that the behavior seems separable, and easier to test and maintain than if we had it all within one gigantic module, which would likely pass 10KLoC on its own.

So the idea of a "batch" module would be to do something like what this structure approximates: create a set of modules that are designed to be included as a set, and have invariants relative to each other. Substrate doesn't support that now, so we do it more ad-hoc here.